### DISCUSSION TITLE
[NOT-BUG] Creo que hay problemas con el PBcreateproperty 2.0.0

### DISCUSSION THREAD

**Discussion by NicolasPLG** [Cotalker/documentation#252] [2. Questions and Answers] [answered]
Hola

Estaba usando el create property con la expression:

```
{
                "_id" : ObjectId("60bfe40f52f19200082446ce"), 
                "key" : "create_arrendatario", 
                "name" : "PBCreateProperty", 
                "version" : "2.0.0", 
                "data" : {
                    "display" : "$VALUE#data|[find=>identifier=arrendatario]|process", 
                    "code" : "$JOIN#_#($VALUE#data|[find=>identifier=arrendatario]|process)#($VALUE#createdAt|[date=>format=DD_MM_YYYY_HH:MM:SS.SSS])", 
                    "propertyType" : "arrendatario"
                }, 
                "next" : {
                    "SUCCESS" : "get_arrendatario", 
                    "ERROR" : ""
                }, 
                "customNetworkRequest" : [

                ]
            }
```

Cuyo error es el siguiente:
```
0:
key: "create_arrendatario"
name: "PBCreateProperty"
status: "ERROR"
result:
errors: Array[1] ["\"display\" must be a string"]
nextBot: ""
iteration: 1
```

Cómo se debe escribir?, probé con [""] y me arrojó otro resultado.

Veré si volvieron por base a la version 1 me funciona

Gracias

**Comment by NicolasPLG** [comment, marked as answer]
Al reemplar a la version 1 me funciona :(

**Reply by guillevalin** [reply]
@NicolasPLG , es porque la versión 2 del bot tiene validadores, y lo que estás haciendo efectivamente está mal.

Tienes:
```"display" : "$VALUE#data|[find=>identifier=arrendatario]|process", ```

Este valor ```$VALUE#data|[find=>identifier=arrendatario]|process``` es un array, y display espera un string. Lo correcto es que pases un elemento del array, ej: ```$VALUE#data|[find=>identifier=arrendatario]|process|0```.

En la versión 1 funciona porque no se validan las entradas, justamente las mejoras de la versión 2 es mejor manejo de las entradas y salidas y manejos de errores.

**Reply by NicolasPLG** [reply]
@guillevalin entiendo!! hay que colocar el numero al cual, quieres agregar. Estaría eso documentado (diferencias entre un bot v1 con el v2 tipo comparativo?

**Reply by guillevalin** [reply]
Es que lo que escribiste antes estaba derechamente malo, pero el bot transparentemente si le pasabas un array, tomaba el primer elemento (el 0). En la versión 2.0 como tenemos validación de entradas, ahora te pide explicitamente un elemento, y no puedes pasarle un array, así que tienes que explicitamente pasarle el elemento 0.

**Reply by NicolasPLG** [reply]
ahora entiendo si o si se debe colocar la expresion |0 para que el bot funcione

gracias por la respuesta :>
